### ISSUE TITLE
Problem compiling with PGI on Summit

### ISSUE THREAD

**Issue by ascheinb** [ECP-copa/Cabana#192]
Getting errors like this:

```
"/ccs/home/scheinberg/Software/Jan20/Cabana/gtest/gtest/gtest-all.cc", line 2:
          error: nonstandard preprocessing directive
  # 1
    ^
```

Even though there is no preprocessing directive on line 2.

And getting warnings like this:

```
"/sw/summit/cuda/10.1.105/bin/../targets/ppc64le-linux/include/crt/host_config.
          h", line 54: warning: unrecognized preprocessing directive
  #warning "crt/host_config.h is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead."
   ^
```

Any ideas? Obvious flag I'm missing?

I do the following setup:
```
module load pgi/19.4
module load cuda/10.1.105
module load cmake/3.14.2
export XGC_PLATFORM=summit
export FC=mpifort
export CC=mpicc
export NVCC_WRAPPER_DEFAULT_COMPILER=mpiCC
export CXX=/ccs/home/scheinberg/Software/Jan20/kokkos/bin/nvcc_wrapper

export KOKKOS_SRC_DIR=/ccs/home/scheinberg/Software/Jan20/kokkos
export KOKKOS_INSTALL_DIR=$KOKKOS_SRC_DIR/cmake_install
export CABANA_INSTALL_DIR=/ccs/home/scheinberg/Software/Jan20/Cabana/install
cmake \
     -D CMAKE_BUILD_TYPE="Release" \
     -D CMAKE_PREFIX_PATH=$KOKKOS_INSTALL_DIR \
     -D CMAKE_INSTALL_PREFIX=$CABANA_INSTALL_DIR \
     -D CMAKE_CXX_COMPILER=$KOKKOS_SRC_DIR/bin/nvcc_wrapper \
     -D Cabana_ENABLE_TESTING=ON \
     -D Cabana_ENABLE_EXAMPLES=ON \
     -D Cabana_ENABLE_Serial=ON \
     -D Cabana_ENABLE_OpenMP=ON \
     -D Cabana_ENABLE_Cuda=ON \
     -D Cabana_ENABLE_MPI=ON \
     -D CMAKE_CXX_FLAGS="-std=c++11" \
     -D KOKKOS_SETTINGS_DIR=$KOKKOS_INSTALL_DIR \
     -D KOKKOS_LIBRARY=$KOKKOS_INSTALL_DIR/lib/libkokkos.a \
     -D KOKKOS_INCLUDE_DIR=$KOKKOS_INSTALL_DIR/include \
     .. ;
make install
```

**Comment by sslattery** [comment]
I'm going to try updating our gtest so it downloads automatically as in Cajita - hopefully this will resolve the issue with an up-to-date gtest.

**Comment by ascheinb** [comment]
Ok, thanks!

btw doesn't look relevant but, for completeness, I also get this warning message when running cmake:

```
CMake Warning (dev) at CMakeLists.txt:79 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable MPI_ROOT is set to:

    /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/pgi-19.4/spectrum-mpi-10.3.0.1-20190611-4ymaahbai7ehhw4rves5jjiwon2laz3a

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by sslattery** [comment]
yeah the `ROOT` variables are being deprecated in CMake - we need to update those as well

**Comment by ascheinb** [comment]
In the meantime, issue "solved" by setting Cabana_ENABLE_TESTING=OFF.

**Comment by streeve** [comment]
Fairly certain this was fixed with #368 - didn't seem to be a continuing issue
